src/plugins/discover/public/services/saved_searches/restore_from_saved_search.ts

```diff
@@ -17,18 +17,14 @@ export const restoreStateFromSavedSearch = ({
   savedSearch: SavedSearch;
   timefilter: TimefilterContract;
 }) => {
-  if (!savedSearch) {
+  if (!savedSearch || !savedSearch.timeRestore) {
     return;
   }
 
-  if (savedSearch.timeRestore && savedSearch.timeRange && isTimeRangeValid(savedSearch.timeRange)) {
+  if (savedSearch.timeRange && isTimeRangeValid(savedSearch.timeRange)) {
     timefilter.setTime(savedSearch.timeRange);
   }
-  if (
-    savedSearch.timeRestore &&
-    savedSearch.refreshInterval &&
-    isRefreshIntervalValid(savedSearch.refreshInterval)
-  ) {
+  if (savedSearch.refreshInterval && isRefreshIntervalValid(savedSearch.refreshInterval)) {
```

Comment:
Time restore logic used to work after reopening the same search or after hitting "Reset search" button

![reset search](https://user-images.githubusercontent.com/1415710/227939807-21472511-985c-43bd-b278-8110eda7feff.gif)

Reply:
Works now, thanks!